### DISCUSSION TITLE
Is it possible to make SwiftGen generate structures out of JSON files?

### DISCUSSION THREAD

**Discussion by mainrs** [SwiftGen/SwiftGen#850] [Q&A] [answered]
```json
// names.json
[
    {
        "name": "a",
        "desc": "b"
    },
    {
        "name": "b",
        "desc": "c"
    }
]
```

```swift
struct Name {
    let name: String
    let desc: String

    init(name: String, desc: String) {
        self.name = name
        self.desc = desc
    }
}

struct JsonFiles {
    let names: [Name] = [
        // ... generated
        Name(name: "a", desc: "b"),
        Name(name: "b", desc: "c")
    ]

    private init() {} 
}
```

Something like this. The current JSON generator seems to rely on `JsonDocument` and would still mean that parsing would happen at runtime, even if it can be pushed to compile-time in a lot of cases by constructing structs directly.

**Comment by djbe** [comment, marked as answer]
The sample templates for JSON, YAML and Plists are only just that, example templates. They work for very simple use cases, but will not match what many users need.

They definitely cannot generate the code you're proposing, because they have to handle much more general cases. 
- Note that there's an `inline-swift5` template for inlining the values like you're proposing, see the [docs for it here](https://github.com/SwiftGen/SwiftGen/blob/stable/Documentation/templates/json/inline-swift5.md).
- From that template you can easily create a custom template that matches your needs. You can find an article that [describes this process here](https://github.com/SwiftGen/SwiftGen/blob/stable/Documentation/Articles/Creating-custom-templates.md).
